spm-monitor/src/main/java/com/sematext/spm/client/jmx/HostOverrideClientSocketFactory.java

```diff
@@ -0,0 +1,26 @@
+package com.sematext.spm.client.jmx;
+
+import java.io.IOException;
+
+import java.rmi.server.RMISocketFactory;
+import java.net.Socket;
+import java.net.ServerSocket;
+
+class HostOverrideClientSocketFactory extends RMISocketFactory {
+    private String host;
+    public HostOverrideClientSocketFactory(String host) {
+        this.host = host;
+    }
+    public ServerSocket createServerSocket(int port) throws IOException {
+        return RMISocketFactory.getDefaultSocketFactory()
+                .createServerSocket(port);
+    }
+    public Socket createSocket(String host, int port) throws IOException {
+        if (host.contains("127.0.0.1") || host.contains("localhost")) {
```

Comment:
I'm hitting issues with JMX communication when the app container is running in a different network, so we have to join it to `st-agent-net`. I think the root cause is basically the same we saw in Cassandra, except the host in RMI stub is set to the IP address of the app container's network. Could we somehow prepare a hack for this, so the host is overridden with a different IP address?

Reply:
Isn't this also always going to be true for BMVM setups? Let's not use HostOverrideClientSocketFactory in such cases.